docs/conf.py

```diff
@@ -336,83 +336,19 @@
-    version_re = re.compile(r'const version = "(?P<version>[^"]*)"')
-    for line in version_file:
-        m = version_re.match(line)
-        if m is None:
-            continue
-        version = m['version']
-        return _major_minor_from_version_string(version), version
-    return None, None
-
-
-def get_tree_juju_version():
-    """Read the version of juju as reported by the juju branch that we're building."""
-    try:
-        # This is the location in Juju 3.6
-        with open('../version/version.go', 'rt') as version_file:
-            major_minor, version = _extract_version_from_version_go(version_file)
-    except FileNotFoundError:
-        # This is the location in Juju 4.0
-        with open('../core/version/version.go', 'rt') as version_file:
-            major_minor, version = _extract_version_from_version_go(version_file)
-    if major_minor is None:
-        raise RuntimeError("could not determine the version of Juju for this directory")
-    return major_minor, version
-
-
-def get_juju_version():
-    """Check to see what version of Juju we are running."""
-    # There is probably more that could be done here about all the possible
-    # version strings juju spits out, but this should cover stripping things
-    # like 'genericlinux' and the architecture out.
-    result = subprocess.run(['juju', 'version'], capture_output=True, text=True)
-    version = result.stdout.rstrip()
-    major_minor = _major_minor_from_version_string(version)
-    if major_minor is None:
-        raise RuntimeError('could not determine version from `juju version`: {}'.format(version))
-    return major_minor, version
-
-
 def generate_cli_docs():
     cli_dir = "reference/juju-cli/"
     generated_cli_docs_dir = cli_dir + "list-of-juju-cli-commands/"
     cli_index_header = cli_dir + 'cli_index'
 
-    tree_major_minor, tree_version = get_tree_juju_version()
-    juju_major_minor, juju_version = get_juju_version()
-    if tree_major_minor != juju_major_minor:
-        warning = ("refusing to rebuild docs with a mismatched minor juju version.\n" +
-                "Found juju {} in $PATH, but the tree reports version {}".format(juju_version, tree_version))
-        print(warning)
-        raise RuntimeError(warning)
-    else:
-        print("generating cli command docs using juju version found in path: {} for tree version {}".format(juju_version, tree_version))
-
     # Remove existing cli folder to regenerate it
     if os.path.exists(generated_cli_docs_dir):
         shutil.rmtree(generated_cli_docs_dir)
 
-    # Generate the CLI docs using "juju documentation" command.
-    subprocess.run(['juju', 'documentation', '--split', '--no-index', '--out', generated_cli_docs_dir],
-                   check=True)
+    # Generate the CLI docs using script.
+    result = subprocess.run(['go', 'run', '../scripts/md-gen/juju-cli-commands/main.go', generated_cli_docs_dir])
```

Comment:
Nice one!

Although this is still quite hacky, this sands down a really sharp corner